system/jlib/jutil.cpp

```diff
@@ -2455,25 +2455,41 @@ jlib_decl bool querySecuritySettings(DAFSConnectCfg *_connectMethod,
              if (*_privateKey)
                  dfsKeywords = true;
         }
+
+        StringBuffer DAFSpassPhraseEnc;
         if (_passPhrase)
         {
             CriticalBlock b(DAFSpassPhraseCrit);
-            if (DAFSpassPhrase.isEmpty())
+            if (DAFSpassPhraseDec.isEmpty())//previously retrieved/decrypted it?
             {
                 const char *passPhrasePtr = conf.queryProp("dfsSSLPassPhrase");
                 if (!isEmptyString(passPhrasePtr))
                 {
+                    DAFSpassPhraseEnc.append(passPhrasePtr);//got encrypted pwd
                     dfsKeywords = true;
-                    decrypt(DAFSpassPhrase, passPhrasePtr);
                 }
             }
-            *_passPhrase = DAFSpassPhrase.str();
         }
 
         if (!dfsKeywords && (_certificate || _privateKey || _passPhrase))
         {
         //end of deprecated code
-            queryHPCCPKIKeyFiles(_certificate, _privateKey, _passPhrase);//use new keywords
+            const char *passPhrasePtr = nullptr;
+            queryHPCCPKIKeyFiles(_certificate, nullptr, _privateKey, _passPhrase ? &passPhrasePtr : nullptr);//use new keywords
+            if (!isEmptyString(passPhrasePtr))
+            {
+                CriticalBlock b(DAFSpassPhraseCrit);
+                if (DAFSpassPhraseEnc.isEmpty())
+                    DAFSpassPhraseEnc.append(passPhrasePtr);//got encrypted pwd
+            }
+        }
+
+        if (_passPhrase)
```

Comment:
Add a comment that legacy querySecuritySettings() returns decrypted passPhrase but new HPCCPKIKeyFiles() returns unaltered passPhrase - so when used here we need to decrypt for compatibility.

Reply:
@mckellyln Comments added